app/component/map/map.cjsx

```diff
@@ -81,20 +75,36 @@ class Map extends React.Component
         fromMarker = <LocationMarker position={origin} className="from"/>
 
       if location.hasPosition == true
-        positionMarker = <Marker position={location.coordinates} icon={Map.currentLocationIcon}/>
+        positionMarker = <PositionMarker/>
 
       if @props.showStops
         stops = <StopMarkerContainer hilightedStops={@props.hilightedStops}/>
 
       vehicles = ""#if @props.showVehicles then <VehicleMarkerContainer/> else ""
 
+      center =
+        if @props.fitBounds
+          undefined #fitBounds is used instead
+        else if @props.lat and @props.lon
+          [@props.lat, @props.lon]
+        else if origin.lat and origin.lon
+          [origin.lat, origin.lon]  #origin is used
+        else
+          undefined # no center to use
+
+      zoom =
```

Comment:
this could be simplified to zoom = if not @props.fitBounds and @props.zoom then @props.zoom